master/buildbot/buildslave/ec2.py

```diff
@@ -120,15 +120,14 @@ def __init__(self, name, password, instance_type, ami=None,
             if aws_id_file_path is None:
                 home = os.environ['HOME']
                 aws_id_file_path = os.path.join(home, '.ec2', 'aws_id')
-            if not os.path.exists(aws_id_file_path):
-                raise ValueError(
-                    "Please supply your AWS access key identifier and secret "
-                    "access key identifier either when instantiating this %s "
-                    "or in the %s file (on two lines).\n" %
-                    (self.__class__.__name__, aws_id_file_path))
-            with open(aws_id_file_path, 'r') as aws_file:
-                identifier = aws_file.readline().strip()
-                secret_identifier = aws_file.readline().strip()
+                if not os.path.exists(aws_id_file_path):
+                    aws_id_file_path = None
+            if aws_id_file_path:
+                log.msg('WARNING: EC2LatentBuildSlave is using deprecated '
```

Comment:
You are still producing a warning only when the `aws_id_file_path` is provided and the corresponding file exists.

More importantly the code at line 120, adds the value if it's not present.

I would put after line 122
```python
else:
     log.msg(BIG FAT WARNING)
```